lib/ansible/modules/cloud/amazon/kinesis_firehose.py

```diff
@@ -0,0 +1,626 @@
+                )
+    else:
+        success = True
+        changed = False
+        err_msg = 'Stream {0} does not exist'.format(delivery_stream_name)
+
+    return success, changed, err_msg, results
+
+
+def to_title(word):
+    return ''.join([si.title() if not (str.isdigit(si[0])) else si for si in word.split('_')])
+
+
+def main():
+    logging.basicConfig(filename='/tmp/kinesis_firehose.log', level=logging.INFO)
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            delivery_stream_name=dict(default=None, required=True),
+            s3_destination_configuration=dict(type='dict'),
+            delivery_stream_type=dict(required=True, type='str', choices=['DirectPut', 'KinesisStreamAsSource']),
+            kinesis_stream_source_configuration=dict(required=True, type='dict'),
+            extended_s3_destination_configuration=dict(type='dict'),
+            wait=dict(default=True, required=False, type='bool'),
+            wait_timeout=dict(default=300, required=False, type='int'),
+            state=dict(default='present', choices=['present', 'absent']),
+        )
+    )
+    mutually_exclusive = [['s3_destination_configuration', 'extended_s3_destination_configuration']]
+
+    required_one_of = [
+        ['s3_destination_configuration', 'extended_s3_destination_configuration']
+    ]
+
+    required_if = [
+        ['state', 'present', ['delivery_stream_type']],
+        ['delivery_stream_type', 'DirectPut', ['s3_destination_configuration']],
+        ['delivery_stream_type', 'KinesisStreamAsSource', ['kinesis_stream_source_configuration']]
+    ]
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        mutually_exclusive=mutually_exclusive,
+        required_if=required_if,
+        required_one_of=required_one_of
+    )
+
+    delivery_stream_name = module.params.get('delivery_stream_name')
+    s3_destination_configuration = module.params.get('s3_destination_configuration')
+    delivery_stream_type = module.params.get('delivery_stream_type')
+    state = module.params.get('state')
+    kinesis_stream_source_configuration = convert_to_title_case(module.params.get('kinesis_stream_source_configuration'))
+    extended_s3_destination_configuration = convert_to_title_case(module.params.get('extended_s3_destination_configuration'))
+    wait = module.params.get('wait')
+    wait_timeout = module.params.get('wait_timeout')
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto3 is required.')
+
+    check_mode = module.check_mode
+    try:
+        region, ec2_url, aws_connect_kwargs = (
+            get_aws_connection_info(module, boto3=True)
+        )
+        client = (
+            boto3_conn(
+                module, conn_type='client', resource='firehose',
+                region=region, endpoint=ec2_url, **aws_connect_kwargs
+            )
+        )
+    except botocore.exceptions.ClientError as e:
```

Comment:
boto3_conn never throws a ClientError at this point.

You need to handle missing Profile (botocore.exceptions.ProfileNotFound) and presumably if `region` is not set too.